### PR TITLE
Create priced button component - Closes #583

### LINKED ISSUE #583: Create a priced primary button
The component should include:
 - A Button component
 - A label saying:
    - "Fee: SAMLE_FEE LSK" if the account has sufficient balance.
    - "Not enough credit to pay SAMLE_FEE LSK fee" if the account has insufficient balance.
 - Should disable button in case of insufficient credit.

### PR DESCRIPTION
**Pull request by reyraa**
Closes #583

### CODE REVIEW

File: src/constants/fees.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default {
+  setSecondPassphrase: 5e8,
+  send: 1e7,
+  registerDelegate: 25e8,
```
Comment (slaweet) [inline src/constants/fees.js]: `vote: 1e8.` is missing

File: src/components/pricedButton/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          <span className={`${styles.fee} ${hasFunds ? '' : styles.error}`}>
+            {
+              hasFunds ? `Fee: ${fromRawLsk(props.fee)} LSK` :
+              `Not enough credit to pay ${fromRawLsk(props.fee)} LSK fee`
+            }
+          </span>
+      }
+      <Button label={props.label}
+        primary={true} raised={true}
+        className='next-button'
```
Comment (slaweet) [inline src/components/pricedButton/index.js]: This should be more flexible, e.g. `className={props.className}`

File: src/constants/fees.js (hunk 1 above)
Comment (reyraa) [inline src/constants/fees.js]: added

File: src/constants/fees.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export default {
+  setSecondPassphrase: 5e8,
+  send: 1e7,
```
Comment (willclarktech) [inline src/constants/fees.js]: Prefer `0.1e8` for uniformity.

File: src/components/pricedButton/pricedButton.css [hunk 4]
```diff
@@ -0,0 +1,11 @@
+.fee {
+  font-size: 12px;
+  line-height: 14px;
+  color: grey;
+  text-align: right;
+  margin: 0 16px;
+  transition: all 0.3s cubic-bezier(0.55, 0, 0.55, 0.2);
+}
+.error {
+  color: #dd2c00 !important;
```
Comment (willclarktech) [inline src/components/pricedButton/pricedButton.css]: Is there a way to avoid the `!important`?

File: src/components/pricedButton/index.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+import sinon from 'sinon';
+import { shallow } from 'enzyme';
+import chaiEnzyme from 'chai-enzyme';
+import Button from 'react-toolbox/lib/button';
+import PricedButton from './index';
+
+chai.use(sinonChai);
+chai.use(chaiEnzyme());
```
Comment (willclarktech) [inline src/components/pricedButton/index.test.js]: Do you guys do this in every file?

File: src/components/pricedButton/index.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+describe('DialogElement', () => {
+  let wrapper;
+  const props = {
+    fee: 5e8,
+    onClick: sinon.spy(),
+  };
+  const insufficientBalance = 0;
```
Comment (willclarktech) [inline src/components/pricedButton/index.test.js]: Prefer the largest possible insufficient balance.

File: src/components/pricedButton/index.test.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      wrapper = shallow(<PricedButton {...props} balance={sufficientBalance} />);
+    });
+
+    it('renders a span saying "Fee: 5 LSK"', () => {
+      expect(wrapper.find('span').text()).to.be.equal('Fee: 5 LSK');
+    });
+
+    it('allows to click on Button', () => {
+      wrapper.find(Button).simulate('click');
+      expect(props.onClick).to.have.been.calledWith();
```
Comment (willclarktech) [inline src/components/pricedButton/index.test.js]: Prefer `.calledWithExactly`.

File: src/components/pricedButton/index.test.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  };
+  const insufficientBalance = 0;
+  const sufficientBalance = 6e8;
+
+  it('renders <Button /> component from react-toolbox', () => {
+    wrapper = shallow(<PricedButton {...props} balance={sufficientBalance} />);
+    expect(wrapper.find(Button)).to.have.length(1);
+  });
+
+  describe('Sufficient credit', () => {
```
Comment (karmacoma) [inline src/components/pricedButton/index.test.js]: `Sufficient funds`

File: src/components/pricedButton/index.test.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      expect(wrapper.find('span').text()).to.be.equal('Fee: 5 LSK');
+    });
+
+    it('allows to click on Button', () => {
+      wrapper.find(Button).simulate('click');
+      expect(props.onClick).to.have.been.calledWith();
+    });
+  });
+
+  describe('Insufficient credit', () => {
```
Comment (karmacoma) [inline src/components/pricedButton/index.test.js]: `Insufficient funds`

File: src/components/pricedButton/index.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import Button from 'react-toolbox/lib/button';
+import { fromRawLsk } from '../../utils/lsk';
+import styles from './pricedButton.css';
+
+const PricedButton = (props) => {
```
Comment (willclarktech) [inline src/components/pricedButton/index.js]: A pattern I like is to destructure props out here for less verbosity in the component body.

File: src/components/pricedButton/pricedButton.css [hunk 11]
```diff
@@ -0,0 +1,11 @@
+.fee {
+  font-size: 12px;
+  line-height: 14px;
+  color: grey;
+  text-align: right;
+  margin: 0 16px;
+  transition: all 0.3s cubic-bezier(0.55, 0, 0.55, 0.2);
+}
```
Comment (karmacoma) [inline src/components/pricedButton/pricedButton.css]: Space between selectors.

File: src/components/pricedButton/index.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  const hasFunds = props.balance >= props.fee;
+  return (
+    <div className='primary-button'>
+      {
+        props.fee &&
+          <span className={`${styles.fee} ${hasFunds ? '' : styles.error}`}>
```
Comment (willclarktech) [inline src/components/pricedButton/index.js]: Prefer parentheses around `span`.

File: src/components/pricedButton/index.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+            {
+              hasFunds ? `Fee: ${fromRawLsk(props.fee)} LSK` :
+              `Not enough credit to pay ${fromRawLsk(props.fee)} LSK fee`
```
Comment (karmacoma) [inline src/components/pricedButton/index.js]: `Insufficient funds for ${fromRawLsk(props.fee)} LSK fee`

File: src/components/pricedButton/index.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+            }
+          </span>
+      }
+      <Button label={props.label}
+        primary={true} raised={true}
```
Comment (willclarktech) [inline src/components/pricedButton/index.js]: Prefer separate lines for separate props.

File: src/components/pricedButton/index.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+        className={`next-button ${props.customClassName}`}
+        disabled={props.disabled || (props.fee && !hasFunds)}
+        onClick={props.onClick.bind(this)}/>
```
Comment (willclarktech) [inline src/components/pricedButton/index.js]: Is binding doing anything here? Seems like it should be the responsibility of whatever is passing the function down.

Comment (karmacoma) [inline src/components/pricedButton/index.js]: `onClick={props.onClick.bind(this)} />`

File: src/components/pricedButton/pricedButton.css (hunk 4 above)
Comment (reyraa) [inline src/components/pricedButton/pricedButton.css]: No need. Preferred is that states override modules. https://smacss.com/

File: src/constants/fees.js (hunk 3 above)
Comment (reyraa) [inline src/constants/fees.js]: done

File: src/components/pricedButton/index.test.js (hunk 6 above)
Comment (reyraa) [inline src/components/pricedButton/index.test.js]: done

File: src/components/pricedButton/index.js (hunk 12 above)
Comment (alepop) [inline src/components/pricedButton/index.js]: Maybe it is a better solution for changing `classnames` based on state https://github.com/JedWatson/classnames

File: src/components/pricedButton/index.test.js (hunk 7 above)
Comment (reyraa) [inline src/components/pricedButton/index.test.js]: This component is not supposed to pass anything to it's click handler. what's the difference?

File: src/components/pricedButton/index.js (hunk 10 above)
Comment (reyraa) [inline src/components/pricedButton/index.js]: done

File: src/components/pricedButton/index.js (hunk 12 above)
Comment (reyraa) [inline src/components/pricedButton/index.js]: done

File: src/components/pricedButton/index.js (hunk 14 above)
Comment (reyraa) [inline src/components/pricedButton/index.js]: done

File: src/components/pricedButton/index.js (hunk 15 above)
Comment (reyraa) [inline src/components/pricedButton/index.js]: binding is the repressibility of presentational component. I believe it's the right place for it.

File: src/components/pricedButton/index.test.js (hunk 8 above)
Comment (reyraa) [inline src/components/pricedButton/index.test.js]: done

File: src/components/pricedButton/index.test.js (hunk 9 above)
Comment (reyraa) [inline src/components/pricedButton/index.test.js]: done

File: src/components/pricedButton/index.js (hunk 13 above)
Comment (reyraa) [inline src/components/pricedButton/index.js]: done

File: src/components/pricedButton/pricedButton.css (hunk 11 above)
Comment (reyraa) [inline src/components/pricedButton/pricedButton.css]: done

File: src/components/pricedButton/index.test.js (hunk 7 above)
Comment (willclarktech) [inline src/components/pricedButton/index.test.js]: screen shot 2017-08-09 at 11 09 25

Comment (reyraa) [inline src/components/pricedButton/index.test.js]: This component is not passing anything. this is about the difference of calling with different arguments.

File: src/components/pricedButton/index.js (hunk 15 above)
Comment (willclarktech) [inline src/components/pricedButton/index.js]: What are you expecting the value of `this` to be?

File: src/components/pricedButton/index.test.js (hunk 7 above)
Comment (reyraa) [inline src/components/pricedButton/index.test.js]: replaced with `calledWithExactly`.

File: src/components/pricedButton/index.test.js (hunk 5 above)
Comment (willclarktech) [inline src/components/pricedButton/index.test.js]: https://github.com/LiskHQ/lisk-nano/issues/589

**Pull request merged**